src/main/java/com/ibm/cloud/cloudant/kafka/transforms/ArrayFlatten.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright © 2022 IBM Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package com.ibm.cloud.cloudant.kafka.transforms;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.data.ConnectSchema;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.apache.kafka.connect.transforms.util.Requirements;
+
+public class ArrayFlatten implements Transformation<SourceRecord> {
+
+    static final String DEFAULT_DELIMITER = ".";
```

Comment:
Will this be configurable in a later version? In particular, thinking of allowed chars in some DB column names which can sometimes be very restrictive.

Reply:
This was listed in my questions for reviewers:
>The default Map flatten transform allows customization of the delimiter, should we also provide this in the array flattener?

I think if we want it we should add it now. It sounds like you are on the "yes" side for that.

Reply:
I'm also +1 for customizing this delimiter.

Reply:
Done in [5b122e5](https://github.com/IBM/cloudant-kafka-connector/pull/93/commits/5b122e53b62ed658589cf04cffd63b9d8ca21236)